Paginated the Lyrics command

Aresiel opened the pull request:
It hopefully works, I can't see any reason why it shouldn't. But then again I don't really do python, so test it maybe? Took heavy inspiration from [help.py](https://github.com/flitzstudios/exorium/blob/d3d6f5b9cfba09acb4e07bbf9090d7d1250fd0f5/utils/help.py#L157) for using paginator.

Aresiel committed:
Paginated the Lyrics command.

It hopefully works, I can't see any reason why it shouldn't. But then again I don't really do python, so test it maybe?

DuckMasterAl commented on cogs/social.py:
```diff
@@ -169,12 +170,17 @@ async def lyrics(self, ctx, *, title):
-
-                await ctx.send(embed=e)
+                
+                lyricsArray = js['lyrics'].split("\n")
```
This splits it for every line. If you wanted to split it between every verse you would have to do `\n\n`.
However, this is not what Flitz intended the code to do. They wanted it to split every 2000 characters (not between lines). I'll fix this issue for you because Flitz said I should fix it.

Aresiel replied:
Would splitting it every 2000 characters not often align with pre-existing letters? And wouldn't the paginator handle showing the appropriate amount of lines?

DuckMasterAl replied:
What I meant by "(not between lines)" is that it should split between the best possible points. Preferably, it would split in between verses.

Aresiel replied:
It's the description of an embed that allows 2048 characters, that's what the paginator uses. Is the wanted behaviour not the paginator show the maximum possible amount of lines then? In which case, wouldn't you want to implement such a feature in the paginator?

TheMoksej replied:
Yes, splitting on every line isn't the best idea. I like the Duck's suggestion for splitting in between verses, but that's still not the best way to do that as verse can be 500 characters and can be 2000 characters, the best way to do this would be to pretty much find the 2000th character and then split the last newline (to make it look better instead of splitting in the middle of the word).

Aresiel replied:
If you aim for as much as possible, yeah, that's the best idea. But wouldn't it often be nicer to have 1 verse per page, considering it's music?

TheMoksej replied:
> If you aim for as much as possible, yeah, that's the best idea. But wouldn't it often be nicer to have 1 verse per page, considering it's music?

This could also be done, however, I dislike this for two reasons. 
1) If the song has 30 verses the paginator would have 30 pages, which would just be annoying to go through all of them.
2) Some songs tend to have 1 or 2 line verses. It'd make the embed look pretty much empty. This is a great example: ![Image](https://cdn.dredd-bot.xyz/VRtWiH)

Aresiel replied:
Ah, most songs I listen to tend to have more paragraph-like verses

Aresiel committed:
Switched to a new way to split the lyrics

The new way tries to fit as many verses into the available 2000 characters as possible. It does this by building up the page 1 verse at a time, until it runs out of space. At which time it adds it to the array of pages and restarts, until it runs out of verses.

Aresiel committed:
Damn indentation

Aresiel committed:
My fault

Aresiel committed:
The linter is good to have

TheMoksej reviewed (changes requested):
This causes the infinite loop which crashes the bot at the end.

Aresiel committed:
Merge branch 'flitzstudios:master' into patch-3

Aresiel committed:
Paginated the lyrics command, it actually works this time. (I know)

Aresiel committed:
Merge remote-tracking branch 'origin/master' into patch-3

# Conflicts:
#	cogs/social.py

Aresiel committed:
I don't even know

Aphidian committed:
slash commands

this doesn't need to be removed as we have slash commands.
